### PR TITLE
Fix assignment of dependency depths (UA-849)

### PR DESCRIPTION
**Pull request by kotodharma**
I think the original code is working, although with a very slight inefficiency. Other than the "tightening up" of the code that makes the diffs hard to read (sorry), the only thing that has really changed is replacing `REPLACE(s.name, '%', '\\%')` inside the EXISTS subquery with just `s.name`, but since the names should not contain `%` anyway, this should not be functionally different. The only real change is the loop condition, which changed from `until current_depth_count == previous_depth_count` to `while current_depth_count > 0`, which should save only 1 wasted loop iteration at the end, but otherwise it should have been working. If the `REPLACE` was intended to do something important algorithmically that I'm missing, please advise.

### CODE REVIEW

**Comment by jonpage** [comment]
Series: 146789, `OCUP%@HAW.M`, is one of many with `%`

File: app/models/series.rb [hunk 1]
```diff
@@ -950,27 +950,25 @@ def Series.assign_dependency_depth
...
+    ActiveRecord::Base.connection.execute(<<~SQL)
       UPDATE series s SET dependency_depth = 1
-      WHERE EXISTS (SELECT 1 FROM data_sources WHERE `dependencies` LIKE CONCAT('% ', s.`name`, '%'));
+      WHERE EXISTS (SELECT 1 FROM data_sources WHERE dependencies LIKE CONCAT('% ', s.name, '%'));
     SQL
-    ActiveRecord::Base.connection.execute(first_level_sql)
     current_depth_count = Series.where(universe: 'UHERO', dependency_depth: 1).count
 
     previous_depth = 1
-    until current_depth_count == previous_depth_count
```
Comment (jonpage) [inline app/models/series.rb]: It's possible for us to have a circular dependency (at least they have existed in the past). In those cases, `previous_depth_count` equals `current_depth_count` once we have eliminated the non-circular dependencies. We could add ` || current_depth_count == 0` to the original `until`.

Comment (jonpage) [reply app/models/series.rb]: If you can guarantee no circular dependencies (i.e., that we will in fact converge to `current_depth_count == 0`, we can go with your solution.

**Comment by kotodharma** [comment]
OK well.... everything else is cosmetic, so I am going to mark this PR as closed.

**Pull request closed without merging**